backend/src/src-licenses/src/main/java/com/siemens/sw360/licenses/db/LicenseDatabaseHandler.java

```diff
@@ -370,7 +372,10 @@ public RequestStatus updateLicense(License inputLicense, User user, User request
 
             License dbLicense = new License();
             if(inputLicense.isSetId()) {
-                dbLicense = licenseRepository.get(inputLicense.getId());
+                License dbLicenseTmp = licenseRepository.get(inputLicense.getId());
```

Comment:
Only that in this case backend does not decide about the id. It must be the shortname...

I think that we should not check the id, but check whether the license is in the db yet. And depending on that add or update.